packages/core-react/src/walletData/index.ts

```diff
@@ -1,47 +1,102 @@
-import { WalletsListEthereum } from '@reef-knot/wallets-list';
-import { WalletAdapterData } from '@reef-knot/types';
+import type { Connector, CreateConnectorFn } from 'wagmi';
 import type { Chain } from 'wagmi/chains';
+import type {
+  WalletAdapterData,
+  WalletAdapterType,
+  WalletConnectorData,
+} from '@reef-knot/types';
 
-let walletDataList: undefined | WalletAdapterData[];
-
-export interface GetConnectorsArgs {
+export interface GetWalletAdaptersListArgs {
+  walletsList: Record<string, WalletAdapterType>;
   rpc: Record<number, string>;
-  chains: Chain[];
   defaultChain: Chain;
   walletconnectProjectId?: string;
 }
 
-export const getWalletDataList = ({
+export const getWalletAdaptersList = ({
+  walletsList,
   rpc,
-  chains,
   defaultChain,
   walletconnectProjectId,
-}: GetConnectorsArgs) => {
-  const walletAdapters = Object.values(WalletsListEthereum);
-  if (!walletDataList) {
-    // Sorting supported chains, so the default chain is always first in the array
-    // Have to do this because WalletConnect v2 via wagmi always tries to connect
-    // to the first chain in the chains array
-    const sortedChains = defaultChain
-      ? chains.sort((chain) => (chain.id === defaultChain.id ? -1 : 1))
-      : chains;
-
-    if (!rpc[defaultChain.id]) {
-      throw 'RPC for default chain must be provided';
-    }
+}: GetWalletAdaptersListArgs) => {
+  const walletAdapters = Object.values(walletsList);
 
-    walletDataList = walletAdapters.map((walletAdapter) =>
-      walletAdapter({
-        rpc,
-        chains: sortedChains,
-        defaultChain,
-        walletconnectProjectId,
-      }),
-    );
+  if (!rpc[defaultChain.id]) {
+    throw 'RPC for default chain must be provided';
   }
-  return walletDataList;
+
+  const walletAdaptersList = walletAdapters.map((walletAdapter) =>
```

Comment:
I don't agree that it is `walletAdaptersList` because it is a list of objects with data returned by wallet adapters.
Wallet adapter is a function, that initializes a wallet and returns some data about the wallet (or I don't know how to name it better than "data").
So, here, the list of wallet adapters is simply `walletAdapters`.

Reply:
The naming is the same as before: https://github.com/lidofinance/reef-knot/blob/main/packages/core-react/src/walletData/index.ts#L33

Reply:
Ok, I think we can rename it in the way it [was named](https://github.com/lidofinance/reef-knot/blob/main/packages/core-react/src/walletData/index.ts#L33) before the update, like walletDataList